app/web/src/store/components.store.ts

```diff
@@ -1180,6 +1179,7 @@ export const useComponentsStore = (forceChangeSetId?: ChangeSetId) => {
               category,
             } = templateData;
 
+            // TODO: @zack should we move this into a WsEvent?? because the changeset in the closure below will be HEAD (if thats where we started) and navigation will fail
```

Comment:
More food for thought @zacharyhamm

Reply:
ahh, yeah that makes sense, yes this should become a ws event